pathod/language/generators.py

```diff
@@ -37,14 +37,19 @@ def __getitem__(self, x):
         d = self.gen.__getitem__(x)
         return self.transform(x, d)
 
-    def __getslice__(self, a, b):
-        d = self.gen.__getslice__(a, b)
-        return self.transform(a, d)
-
     def __repr__(self):
         return "'transform(%s)'" % self.gen
 
 
+def rand_byte(chars):
+    """
+        Return a random character as byte from a charset.
+    """
+    # bytearray has consistent behaviour on both Python 2 and 3
+    # while bytes does not
+    return bytes(bytearray([random.choice(chars)]))
```

Comment:
I really appreciate the work that went into this, nonetheless let me propose to simplify it by keeping in mind that we want to ditch Python 2 completely once your awesome job is done here:
```
if six.PY2:
    return random.choice(chars)
return bytes([chars])
```